src/dune_engine/build_config.ml

```diff
@@ -110,10 +110,19 @@ let set
         contexts
         ~f:(fun ((ctx : Build_context.t), ctx_type) -> ctx.name, (ctx, ctx_type)))
   in
-  let () =
+  let cache_config =
     match (cache_config : Dune_cache.Config.t) with
-    | Disabled -> ()
-    | Enabled _ -> Dune_cache_storage.Layout.create_cache_directories ()
+    | Disabled -> cache_config
+    | Enabled _ ->
+      (match Dune_cache_storage.Layout.create_cache_directories () with
```

Comment:
Can you move this check to the caller of this function? This should be either in the rules or in the command line (if my memory serves correctly)

Reply:
I did that in 984f8904b but that means that the call to `Dune_cache_storage.Layout.create_cache_directories` now lives in the `dune_rules`. Is that the right place?

I didn't want to make `Build_config.set` return an error that the caller has to understand and call `set` again with the cache disabled.

Reply:
> I did that in https://github.com/ocaml/dune/commit/984f8904b790de94abe8bf4d445b6d6d5428d3e6 but that means that the call to Dune_cache_storage.Layout.create_cache_directories now lives in the dune_rules. Is that the right place?

Ideally, it would live in the command line but having it live in the rules is acceptable for now.

> I didn't want to make Build_config.set return an error that the caller has to understand and call set again with the cache disabled.

Yeah, `Build_config.set` should never fail. It is only used to set some hooks to configure the engine and the rules.